velox/connectors/hive/tests/HiveDataSinkTest.cpp

```diff
@@ -505,6 +510,39 @@ TEST_F(HiveDataSinkTest, basic) {
   verifyWrittenData(outputDirectory->path);
 }
 
+TEST_F(HiveDataSinkTest, parquetBlockSizeAndRows) {
+  std::unordered_map<std::string, std::string> connectorConfig;
+  connectorConfig.emplace("parquet_optimized_writer_max_block_size", "10L");
+  connectorConfig.emplace("parquet_optimized_writer_max_block_rows", "100");
+
+  connectorConfig_ = std::make_shared<HiveConfig>(
+      std::make_shared<core::MemConfig>(std::move(connectorConfig)));
+
+  const auto outputDirectory = TempDirectoryPath::create();
+  auto dataSink = createDataSink(
+      rowType_, outputDirectory->path, dwio::common::FileFormat::PARQUET);
+
+  const int numBatches = 10;
+  const auto vectors = createVectors(500, numBatches);
+  for (const auto& vector : vectors) {
+    dataSink->appendData(vector);
+  }
+  dataSink->close();
+
+  dwio::common::ReaderOptions readerOpts{pool_.get()};
+  const std::vector<std::string> filePaths = listFiles(outputDirectory->path);
+  auto bufferedInput = std::make_unique<dwio::common::BufferedInput>(
+      std::make_shared<LocalReadFile>(filePaths[0]),
+      readerOpts.getMemoryPool());
+
+#ifdef VELOX_ENABLE_PARQUET
+  auto reader = std::make_unique<facebook::velox::parquet::ParquetReader>(
+      std::move(bufferedInput), readerOpts);
+  EXPECT_EQ(reader->numberOfRowGroups(), 50);
```

Comment:
@JkSelf Can you explain how the params RowGroup size 10 and RowGroup rows 100 translate to this output? This does not verify the individual RowGroup values.
We should add an API to get the Parquet File Metadata, so that we can get the RowGroups and ColumnChunks metadata.

Reply:
I will work on this.

Reply:
https://github.com/facebookincubator/velox/issues/8485